pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -27,24 +29,52 @@ type Adapter struct {
 	Client lclient.Client
 	common.AdapterContext
 
-	componentAliasToVolumes   map[string][]adaptersCommon.DevfileVolume
-	uniqueStorage             []adaptersCommon.Storage
+	componentAliasToVolumes   map[string][]common.DevfileVolume
+	uniqueStorage             []common.Storage
 	volumeNameToDockerVolName map[string]string
+	devfileBuildCmd           string
+	devfileRunCmd             string
+	supervisordVolumeName     string
 }
 
 // Push updates the component if a matching component exists or creates one if it doesn't exist
 func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	componentExists := utils.ComponentExists(a.Client, a.ComponentName)
 
 	// Process the volumes defined in the devfile
-	a.componentAliasToVolumes = adaptersCommon.GetVolumes(a.Devfile)
+	a.componentAliasToVolumes = common.GetVolumes(a.Devfile)
 	a.uniqueStorage, a.volumeNameToDockerVolName, err = storage.ProcessVolumes(&a.Client, a.ComponentName, a.componentAliasToVolumes)
 	if err != nil {
 		return errors.Wrapf(err, "Unable to process volumes for component %s", a.ComponentName)
 	}
+	a.devfileBuildCmd = parameters.DevfileBuildCmd
+	a.devfileRunCmd = parameters.DevfileRunCmd
+
+	// Validate the devfile build and run commands
+	log.Infof("\nValidation")
+	pushDevfileCommands, err := common.ValidateAndGetPushDevfileCommands(a.Devfile.Data, a.devfileBuildCmd, a.devfileRunCmd)
+	if err != nil {
+		return errors.Wrap(err, "failed to validate devfile build and run commands")
+	}
+	log.Successf("Devfile validated")
+
+	// Get the supervisord volume
+	supervisordLabels := utils.GetSupervisordVolumeLabels()
+	supervisordVols, err := a.Client.GetVolumesByLabel(supervisordLabels)
```

Comment:
nitpick, but maybe use full names for parameters? `supervisordVolumes` would be okay